### PR TITLE
tests: increase approx size equal threshold to avoid `test_lsn_lease_size` flakiness

### LINKED ISSUE #8293: test_lsn_lease_size flaky
https://neon-github-public-dev.s3.amazonaws.com/reports/pr-8291/9811816038/index.html#suites/d70040d097c5c7c438e421e0416d69ce/8b9d1e05e6c02925

```
AssertionError: assert 312497288 == 312316992 ± 3.3e+04
  comparison failed
  Obtained: 312497288
  Expected: 312316992 ± 3.3e+04
```

cc @yliang412 do you have any idea or I/we can look into it next week?

### PR DESCRIPTION
**Pull request by yliang412**
## Summary of changes

Increase the `assert_size_approx_equal` threshold to avoid flakiness of `test_lsn_lease_size`. Still needs more investigation to fully resolve #8293.

- Also set `autovacuum=off` for the endpoint we are running in the test.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Comment by yliang412** [comment]
Followup: on PG16, it's much larger ([allure](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-8254/9810252494/index.html#suites/d70040d097c5c7c438e421e0416d69ce/eebc57a5953a6808)).

File: test_runner/regress/test_tenant_size.py [hunk 1]
```diff
@@ -486,9 +486,9 @@ def assert_size_approx_equal(size_a, size_b):
     """
 
     # Determined empirically from examples of equality failures: they differ
-    # by page multiples of 8272, and usually by 1-3 pages.  Tolerate 4 to avoid
+    # by page multiples of 8272, and usually by 1-4 pages.  Tolerate 5 to avoid
     # failing on outliers from that observed range.
-    threshold = 4 * 8272
+    threshold = 5 * 8272
```
Comment (arpad-m) [inline test_runner/regress/test_tenant_size.py:491]: could you maybe increase it even further so that the test gets even less flaky?

**Review by arpad-m — APPROVED** [review]
good as a temporary workaround, later we can find a better fix

**Review by skyzh — APPROVED** [review]

File: test_runner/regress/test_tenant_size.py (hunk 1 above)
Comment (yliang412) [reply test_runner/regress/test_tenant_size.py]: Increased to a much larger threshold.

**Pull request merged**